esp/services/ws_cloud/ws_cloudService.cpp

```diff
@@ -59,6 +65,32 @@ bool CWsCloudEx::onGetPODs(IEspContext& context, IEspGetPODsRequest& req, IEspGe
     return true;
 }
 
+const char* CWsCloudEx::readPublicServices(const char* allServices, StringBuffer& publicServices)
+{
+    ICopyArrayOf<IPropertyTree> notPublic;
```

Comment:
Instead of keeping track of which objects are not public and then removing them, I think it would be easier and clearer if you treated this whole function as a JSON string builder function.  

Immediately start building the json
  1. build the begining json of the publicServices string.
  2. create the servicesItr, just as you do now, then
  3. ForEach service
  4.   check if it is public
  5.       if no, continue
            if yes,
             {
                delimitJSON(publicServices);
                toJSON(serviceTree, publicServices);
              }

6. end ForEach
7. build the closing JSON to finish the object

Should be easier to read, and nothing to delete/clean up.